contracts/modules/ProtocolSettings/ProtocolSettings.sol

```diff
@@ -136,8 +136,12 @@ contract ProtocolSettings is State, ProtocolSettingsEvents {
                 0x4a99e3a1, // setSwapApprovals(address[])
                 addrs
             );
-            (bool success,) = swapsImpl.delegatecall(data);
-            require(success, "approval calls failed");
+            IDexRecords records = IDexRecords(swapsImpl);
```

Comment:
T: this is for convenience. however we can just call setSwapApprovals(address[]) on swap implementation directly. do we really need this here?

Reply:
It is pretty convenient to have and doesn't need to be triggered so I do not see the harm of it